core/src/main/java/io/kestra/core/docs/AbstractClassDocumentation.java

```diff
@@ -80,7 +88,14 @@ protected AbstractClassDocumentation(JsonSchemaGenerator jsonSchemaGenerator, Cl
 
     protected static Map<String, Object> flatten(Map<String, Object> map, List<String> required) {
         map.remove("type");
-        return flatten(map, required, null);
+        return flatten(map, required, (String) null);
+    }
+
+    protected static Map<String, Object> flatten(Map<String, Object> map, List<String> required, Boolean keepType) {
+        if (!keepType) {
+            map.remove("type");
+        }
+        return flatten(map, required, (String) null);
```

Comment:
Cast is necessary because we have now two `flatten` method with 3 params that could works with a null as a 3rd parameter